stateful/stateful.go

```diff
@@ -0,0 +1,105 @@
+// Package stateful provides stateful HTTP forwarding (i.e. maintaining a 1 to 1
+// relationship between downstream and upstream connections). Clients wishing to
+// take advantage of this capability need to send an initial GET request
+// (analogous to a CONNECT request) that includes the desired host and the HTTP
+// header "X-Lantern-Stateful: true".
```

Comment:
CONNECT means that we establish a dumb tunnel that simply forwards bytes. This stateful forwarder actually interprets the HTTP protocol, which is different.